dependency-resolution/index.md

```diff
@@ -59,3 +59,25 @@ The rough list is then simplified, by keeping only the newest version of any lis
 Hence, `Dx` which has the highest `semver` version is selected and that is `Dx v1.5.0`.
 
 By following this algorithm, the build list is guaranteed to include the oldest module versions available that meet the requirements, thus achieving reproducible behavior.
+
+## Node.js + npm
+
+[npm](https://docs.npmjs.com/about-npm) is a package manager for [Node.js](https://nodejs.org/en), it facilitates the installation of dependencies for use by Node.js.
+To understand what dependency is used at runtime we need to understand both npm and Node.js work.
+
+As a runtime for an interpreted language, Node.js resolves dependencies at runtime.
+To do this it starts to look for a folder structure matching the dependency name inside a `node_modules/` directory in the directory of the file importing the dependency.
```

Comment:
Do you mean it recursively goes up the filesystem tree to look for `node_modules` and then searching inside it for dependency tree?

Reply:
If yes, a `tree`-like output would be very nice to explain it.

Reply:
> Do you mean it recursively goes up the filesystem tree to look for node_modules and then searching inside it for dependency tree?
>
> If yes, a tree-like output would be very nice to explain it.

Yes, tried to clarify this a bit more and added an example. Let me know if you have further suggestions.

Reply:
It is clear now. Thanks!